aten/src/ATen/native/cuda/Reduce.cuh

```diff
@@ -35,6 +35,25 @@ static inline int last_pow2(int n) {
   return std::max(1, n - (n >> 1));
 }
 
+// returns reduced fraction numerator & denominator
+C10_HOST_DEVICE static void reduce_fraction(size_t &numerator, size_t &denominator) {
+  // get GCD of num and denom using Euclid's algorithm.
+  // Can replace this with std::gcd if we ever support c++17.
+  size_t a = denominator;
+  size_t b = numerator;
+  while (b != 0) {
+      a %= b;
+      // swap(a,b)
+      a ^= b;
+      b ^= a;
+      a ^= b;
```

Comment:
IIRC XOR swaps are not really the fastest way to do this. Can't you just `std::swap` which is so much more readable?

Reply:
See [the commentary on the Wikipedia article](https://en.wikipedia.org/wiki/XOR_swap_algorithm#Reasons_for_avoidance_in_practice).

Reply:
IIRC, there's no equivalent device function for std::swap.

For the equivalent issue, I could update the condition check earlier to ensure that this function is safe (also proper comment as well)

Reply:
I thought `std` works in device functions, doesn't it? Otherwise, let's just use a temporary.